src/User/AccountActivationMailerTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User;
+
+use Flarum\Mail\Job\SendRawEmailJob;
+
+trait AccountActivationMailerTrait
+{
+    /**
+     * @param User $user
+     * @param string $email
+     * @return EmailToken
+     */
+    protected function generateToken(User $user, $email)
```

Comment:
I'm leaning towards keeping it like this, with the email passed in separately to allow future flexibility if we decide to adopt multiple emails per user.